File: app/javascript/channels/room_channel.js
import consumer from "channels/consumer"

// appRoomという変数に格納
const appRoom = consumer.subscriptions.create("RoomChannel", {
  connected() {
    // Called when the subscription is ready for use on the server
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    console.log("receivedしたよー")
    // Called when there's incoming data on the websocket for this channel
    return alert(data["message"])
  },

  speak: function(message) {
    console.log("JavaScript側のspeakへ来たよー")
    return this.perform('speak', {message: message});
  }
});

window.document.onkeydown = function(event) {
  if(event.key == "Enter") {
    console.log("Enterを押したよー")
    appRoom.speak(event.target.value);
    event.target.value = "";
    event.preventDefault();
  }
}

File: app/channels/room_channel.rb
  class RoomChannel < ApplicationCable::Channel
    def subscribed
      p "subscribedしたよー"
      # console.log("subscribedしたよー")
      stream_from "room_channel"
    end
  
    def unsubscribed
      # Any cleanup needed when channel is unsubscribed
    end
  
    def speak(data)
      p "Channel側のspeakへ来たよー"
      # ActionCable.server.broadcast("room_channel", {message: data["message"]})
      Message.create! content: data["message"]
    end
  end
